packages/rum-core/src/domain/rumEventsCollection/view/trackViewMetrics.ts

```diff
@@ -8,6 +19,70 @@ import type { LifeCycle } from '../../lifeCycle'
 import { LifeCycleEventType } from '../../lifeCycle'
 import { waitPageActivityEnd } from '../../waitPageActivityEnd'
 
+import { getScrollY } from '../../../browser/scroll'
+import { getViewportDimension } from '../../../browser/viewportObservable'
+
+export interface ScrollMetrics {
+  maxScrollDepth?: number
+  maxScrollHeight?: number
+  maxScrollDepthTime?: Duration
+  maxScrollTop?: number
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think the scroll metrics don't have to be optional.

```suggestion
export interface ScrollMetrics {
  maxScrollDepth: number
  maxScrollHeight: number
  maxScrollDepthTime: Duration
  maxScrollTop: number
}
```